src/xenia/vfs/virtual_file_system.cc

```diff
@@ -307,5 +310,74 @@ X_STATUS VirtualFileSystem::OpenFile(Entry* root_entry,
   return result;
 }
 
+int VirtualFileSystem::ExtractFiles(Device* device,
+                                    std::filesystem::path base_path) {
+  // Run through all the files, breadth-first style.
+  std::queue<vfs::Entry*> queue;
+  auto root = device->ResolvePath("/");
+  queue.push(root);
+
+  // Allocate a buffer when needed.
+  size_t buffer_size = 0;
+  uint8_t* buffer = nullptr;
+
+  while (!queue.empty()) {
+    auto entry = queue.front();
```

Comment:
I can't find the progress bar code anywhere (wierd), but you could probably just change the title bar saying the percentage of the queue that's completed at the beginning of this loop.